Mathlib/Algebra/Homology/HomologicalComplex.lean

```diff
@@ -92,6 +92,37 @@ theorem ext {C₁ C₂ : HomologicalComplex V c} (h_X : C₁.X = C₂.X)
   · rw [s₁ i j hij, s₂ i j hij]
 #align homological_complex.ext HomologicalComplex.ext
 
+/-- The obvious isomorphism `K.X p ≅ K.X q` when `p = q`. -/
+def XIsoOfEq (K : HomologicalComplex V c) {p q : ι} (h : p = q) :
+  K.X p ≅ K.X q := eqToIso (by rw [h])
+
+@[simp]
+lemma XIsoOfEq_rfl (K : HomologicalComplex V c) (p : ι) :
+  K.XIsoOfEq (rfl : p = p) = Iso.refl _ := rfl
+
+@[reassoc (attr := simp)]
+lemma XIsoOfEq_trans_hom (K : HomologicalComplex V c) {p₁ p₂ p₃ : ι}
```

Comment:
Based on the name, I would have expected `(K.XIsoOfEq (h₁₂.trans h₂₃)).hom` on the LHS instead of the RHS. Perhaps the lemma should be called `XIsoOfEq_hom_comp_XIsoOfEq_hom`? Also, should there be three more variants of this lemma involving different combinations of `hom` and `inv`?

Reply:
Thanks for the review! I have just added these `hom/inv` variants.